### PR TITLE
fix: cannot save assign control max count

### PR DESCRIPTION
**Pull request by mkoontz-rewired**
## Description

Fixing an issue that showed up in helpscout and fixing other small issues in the same area. See commit messages for more detail.

## Motivation and Context

- Root cause fix for https://secure.helpscout.net/conversation/2252793441/8846/ https://rewiredcoop.slack.com/archives/C02L56CP92A/p1684939561227669
- Also realized there's no way to clear escalation tags while fixing this, so fixed that, too. 
- Some small styling changes, see screenshot

## How Has This Been Tested?

locally

## Screenshots (if appropriate):
OLD:
Screen Shot 2023-05-24 at 2 49 24 PM

NEW:
Screen Shot 2023-05-24 at 2 43 44 PM

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have included updates for the documentation accordingly.

### CODE REVIEW

File: src/components/QuickSearchToolbar.tsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import React from "react";
 
 interface QuickSearchToolbarProps {
   value: string;
-  onChange(): void;
+  onChange: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>;
```
Comment (hiemanshu) [inline src/components/QuickSearchToolbar.tsx:9]: Question: Is this a related change? If not, should be removed.

File: src/containers/AdminAssignmentControl/components/AssignmentRow.tsx [hunk 2]
```diff
@@ -37,17 +41,18 @@ const AssignmentRow: React.FC<Props> = (props) => {
...
     onChange({
       maxRequestCount: maxRequestCount ? parseInt(maxRequestCount, 10) : null
     });
+  };
 
   const handleChangeEscalationTags = (
     _event: React.ChangeEvent<any>,
-    value: Array<TagWithTitle> | null
-  ) => {
+    value: TagWithTitle[]
```
Comment (hiemanshu) [inline src/containers/AdminAssignmentControl/components/AssignmentRow.tsx:54]: nit: I prefer the `Array<type>` over `Type[]` style, reads better. Don't have to make that change, but we should settle on a convention 😓

**Review by hiemanshu — APPROVED** [review]
LGTM

File: src/containers/AdminAssignmentControl/components/AssignmentRow.tsx (hunk 2 above)
Comment (ajohn25) [reply src/containers/AdminAssignmentControl/components/AssignmentRow.tsx:54]: i like `Type[]` lol we should vote

File: src/components/QuickSearchToolbar.tsx (hunk 1 above)
Comment (mkoontz-rewired) [reply src/components/QuickSearchToolbar.tsx:9]: It's related to me. The max count bug was caused by switching to @material-ui/core/TextField from material-ui/TextField, which changed the typing of onChange. 

I was looking through other instances of onChange used with <TextField> and saw that this one had bad typing. Feels related enough to the core issue involved.

File: src/containers/AdminAssignmentControl/components/AssignmentRow.tsx (hunk 2 above)
Comment (mkoontz-rewired) [reply src/containers/AdminAssignmentControl/components/AssignmentRow.tsx:54]: yeah fine with either but I also like Type[] lol

**Review by ajohn25 — APPROVED** [review]
